src/ContextualPopup/ContextualPopup.js

```diff
@@ -538,18 +557,37 @@ module.exports = kind(
 	/**
 	* @private
 	*/
-	showHideScrim: function (inShow) {
+	updateScrim: function (old, value, source) {
+		if (this._updateScrimMutex) return;
+		this._updateScrimMutex = true;
+
+		// We sync modal and spotlightModal here because binding doesn't guarantee sequence
+		// when it is used with observers.
+		if (source == 'modal') this.set('spotlightModal', this.modal);
+		if (source == 'spotlightModal') this.set('modal', this.spotlightModal);
+		this.showHideScrim(this.showing);
+
+		this._updateScrimMutex = false;
+	},
+
+	/**
+	* @private
+	*/
+	showHideScrim: function (show) {
 		if (this.floating && (this.scrim || (this.modal && this.scrimWhenModal))) {
-			var scrim = this.getScrim();
-			if (inShow && this.modal && this.scrimWhenModal) {
+			this._scrim = this.getScrim();
+			if (show) {
```

Comment:
Would it work to combine this with the conditional above, so that we have a single if/else for showing/hiding the scrim? We could consolidate all of the clean-up logic and remove some redundancy. Also, I don't know if we necessarily need to cache the scrim reference as an instance member, as the scrim is a singleton anyways.